src/ant/src/AntennaChecker.cc

```diff
@@ -1744,12 +1750,14 @@ bool AntennaChecker::checkViolation(const PARinfo& par_info, dbTechLayer* layer)
     }
 
     const double PSR_ratio = antenna_rule->getPSR();
+    const dbTechLayerAntennaRule::pwl_pair diffPSR = antenna_rule->getDiffPSR();
+    const double diffPSR_ratio = getPwlFactor(diffPSR, diff_area, 0.0);
+
+
     if (PSR_ratio != 0) {
       if (psr > PSR_ratio)
         return true;
     } else {
-      dbTechLayerAntennaRule::pwl_pair diffPSR = antenna_rule->getDiffPSR();
-      const double diffPSR_ratio = getPwlFactor(diffPSR, diff_area, 0.0);
```

Comment:
Is this change needed?

Reply:
Not really. It was introduced when the checks were handled here instead of the init function but it is not required anymore. It should not have any impact though.

Reply:
There would be a small runtime cost and a bit of readability.  If there is no reason then let's return it to the previous form.